src/Http/Controllers/Configuration/UserController.php

```diff
@@ -42,9 +46,74 @@ class UserController extends Controller
     public function getAll()
     {
 
-        $groups = $this->getAllGroups();
+        if(\request()->ajax()){
 
-        return view('web::configuration.users.list', compact('groups'));
+            $groups = $this->getAllFullUsers();
+
+            return DataTables::of($groups)
+                ->editColumn('refresh_token', function ($row) {
+                    return view('web::configuration.users.partials.refresh-token', compact('row'));
+                })
+                ->editColumn('name', function ($row) {
+
+                    if($row->id === 1)
+                        return 'Admin';
+
+                    $character = CharacterInfo::find($row->id) ?: $row->id;
+
+                    return view('web::partials.character', compact('character'));
+                })
+                ->editColumn('last_login', function ($row) {
+                    return human_diff($row->last_login);
+                })
+                ->editColumn('email', function ($row) {
+                    if(empty($row->email))
+                        return trans('web::seat.no_email');
+
+                    return $row->email;
+                })
+                ->addColumn('action_buttons', function ($row) {
+                    return view('web::configuration.users.partials.action-buttons', compact('row'));
+                })
+                ->addColumn('roles', function (User $user) {
+                    return $user->group->roles->map(function ($role) {
```

Comment:
Is there a way you return the caption `No role defined` as you did for email ?

Reply:
done